### PR TITLE
config: add support for port '*'

### LINKED ISSUE #57: userInfoMatcher add support for port '*'
Related work https://github.com/saucelabs/libsauceconnect/commit/d9c5a5436d9378f5417524ff59b16391976296e9

### PR DESCRIPTION
**Pull request by Choraden**
Parsing user:password@host:port string will check if port is '*' and change it to '0'.

Fixes #57

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -51,9 +51,14 @@ func ParseHostPortUser(val string) (*HostPortUser, error) {
 		return nil, err
 	}
 
+	p := u.Port()
```
Comment (mmatczuk) [inline config.go:54]: Isn't adding it to CredentialsManager nicer and simpler?

Comment (mmatczuk) [reply config.go:54]: Also, a test would be good, CredentialsManager has the tests to add this

File: config.go [hunk 2]
```diff
@@ -43,10 +43,18 @@ func validatedUserInfo(ui *url.Userinfo) error {
...
+		s[len(s)-1] = "0"
+	}
+	return strings.Join(s, ":")
+}
+
 // ParseHostPortUser parses a user:password@host:port string into HostUser.
 // User and password cannot be empty.
 func ParseHostPortUser(val string) (*HostPortUser, error) {
-	u, err := url.Parse("http://" + val)
+	u, err := url.Parse("http://" + wildcardPortTo0(val))
```
Comment (Choraden) [inline config.go:57]: I'm afraid, we can't move it to credentials matcher. 
url.Parse fails on port '*'.

Comment (mmatczuk) [reply config.go:57]: Gotcha.

**Pull request merged**